src/agoradatatools/etl/transform/immunohisto_transform.py

```
@@ -65,6 +67,149 @@ def prepare_immunohisto_data(df: pd.DataFrame) -> pd.DataFrame:
+    placeholder entries.
+
+    Example:
+        If the dataset has ages 4, 8, and 12 months but one tissue is missing 8 month
+        data, this function adds rows to fill in the missing time point with
+        entries containing empty data arrays and units.
+
+    Args:
+        data_rows: DataFrame with grouped data including columns for name, evidence_type,
+                   tissue, age, units, y_axis_max, and data
+
+    Returns:
+        Updated DataFrame with missing age entries added
+    """
+    # Get all unique ages that exist in the data
+    available_ages = list(data_rows["age"].drop_duplicates())
+
+    # All unique combinations of groups (name, evidence_type, tissue, y_axis_max)
+    fill_df = (
+        data_rows[["name", "evidence_type", "tissue", "y_axis_max"]]
+        .copy()
+        .drop_duplicates()
+    )
+
+    # Make an "age" column where each entry is a list of all possible ages
+    fill_df["age"] = [available_ages] * fill_df.shape[0]
+
+    # "explode" makes one row per age + group. Then merge back into the data to create new rows for missing ages
+    fill_df = fill_df.explode("age").merge(
+        data_rows, how="outer", validate="one_to_one"
+    )
+
+    # Fill NA values for units. Can't use fillna to make an empty list so we add an extra line
+    fill_df = fill_df.fillna({"units": ""})
+    fill_df["data"] = fill_df["data"].apply(lambda x: [] if x is np.nan else x)
+
+    return fill_df
+
+
+def _extract_age_num(age_str: str) -> float:
+    """
+    Extract numeric age value for sorting purposes.
+
+    Args:
+        age_str: A string containing the age value (e.g., "12 months")
+
+    Returns:
+        float: The numeric age value, or inf for invalid ages
+    """
+    try:
+        numeric_value = float(age_str.split()[0])
+        return numeric_value
+    except (ValueError, IndexError, AttributeError):
+        # For invalid ages, use a large number for consistent sorting
+        return float("inf")
```

To double-check: @JessterB if the age field is mis-formed (i.e. "6months" instead of "6 months") or missing values, do we want to throw an error, or return `inf` as written? I'm a little worried about it failing silently with `inf` and us not catching it.

---

Throwing an error here seems reasonable; I'd also expect GX validation to fail on malformed age strings, so I don't expect we would miss the issue entirely once we add GX suites.

---

I'll throw an error! I did the inf so that we could still get a result and use age for sorting.